core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java

```diff
@@ -192,14 +192,10 @@ private YarnStackV4Parameters getYarnProperties(DistroXV1Request source, Detaile
     public DistroXV1Request convert(StackV4Request source) {
         DistroXV1Request request = new DistroXV1Request();
         request.setName(source.getName());
-        DetailedEnvironmentResponse environment = null;
         if (source.getEnvironmentCrn() != null) {
-            environment = environmentClientService.getByCrn(source.getEnvironmentCrn());
-            if (environment != null && environment.getEnvironmentStatus() != EnvironmentStatus.AVAILABLE) {
-                throw new BadRequestException(String.format("Environment state is %s instead of AVAILABLE", environment.getEnvironmentStatus()));
-            }
+            Optional<DetailedEnvironmentResponse> environment = getEnvironmentName(source.getEnvironmentCrn());
```

Comment:
if the source has no env crn why we woul like to query it ?

Reply:
do we? in line 194 we have a check for the env crn. if it's null then no query happens, otherwise, we set the name based on the result of line 214

Reply:
now I see that part thanks